sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -845,4 +845,24 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
     val outputOuter = Row.fromSeq((1 to N).map(_ => outputInner))
     checkEvaluation(cast(Literal.create(inputOuter, fromOuter), toOuter), outputOuter)
   }
+
+  test("SPARK-22570: should not create a lot of instance variables") {
```

Comment:
In the future we may fold global variables to arrays, and these tests become useless. I think a more reliable way is: check the number of global variables in `CodegenContext`

Reply:
Could you elaborate this thought?  
Do you want to this new check in test cases? Or do you mean we will remove these tests for global variables since we will have new check in another PR?